ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/MiniProtocol/ChainSync/Client.hs

```diff
@@ -195,6 +197,12 @@ bracketChainSyncClient ::
     -- It's more important that the client is removed from the set promptly
     -- than it is for the client to be added promptly, because of how this is
     -- used by the GSM to determine that the node is done syncing.
+ -> (STM m GsmState,
+     StrictTVar m (Map peer (GsmState -> m ())))
```

Comment:
Yeah, this looks good; with #1019, it seems natural to store these callbacks (or maybe (equivalently) the LoP buckets directly) in `ChainSyncClientHandle`/`ChainSyncState` as introduced there.

Reply:
Done in 5807d4cb27d75ab5d0327812c8ff5551b2c899ae.